plugins/newsdownloader.koplugin/main.lua

```diff
@@ -74,7 +74,14 @@ function NewsDownloader:addToMainMenu(menu_items)
         sub_item_table = {
             {
                 text = _("Download news"),
-                callback = function() self:loadConfigAndProcessFeeds() end,
+                callback = function()
+                    if not NetworkMgr:isOnline() then
+                        wifi_enabled_before_action = false
+                        NetworkMgr:turnOnWifiAndWaitForConnection(self.loadConfigAndProcessFeeds)
```

Comment:
This should be `NetworkMgr:promptWifiOn(self.loadConfigAndProcessFeeds)`

Reply:
The goal was to do the same thing with using less tapping/clicking. 

On my emulator `NetworkMgr:promptWifiOn(self.loadConfigAndProcessFeeds)` does nothing once I clicked "turn on wifi". Does it work correclty on real device? 
And `NetworkMgr:turnOnWifiAndWaitForConnection(self.loadConfigAndProcessFeeds)` waits for internet connection (timeout 60sec) and then start processing. This works fine on both emulator & android.

ps: I didn't understand if there was something wrong on your video with Kobo. With my change I give 60 sec timeout so system can display available wifi networks and user have that time to select network, maybe type password, and wait for establish connection. Tested on real device (Android) works fine.

Reply:
The point is that this is under the hood functionality that should not be called directly. Like I already said, everything from https://github.com/koreader/koreader/pull/3395#issuecomment-338458570 applies and then some. Individual parts of the program should not introduce random behavior like that.

Reply:
PS. Everything's good on the video. But it wouldn't be with useless extra messages.

Reply:
Incidentally, doesn't this logic make more sense where it was before? This way everything calling it would have to make sure it enables wifi first, whereas otherwise you can safely just call `loadConfigAndProcessFeeds()` and it'll take care of the rest.